rework_pysatl_mpest/preprocessing/components_family/criterions/sample_criterions.py

```diff
@@ -0,0 +1,175 @@
+        median = np.median(X)
+        result = np.log((xmax - median) / (median - xmin))
+        return result if not np.isnan(result) else 0
+
+
+class COutlierFraction(ASampleClassifierCriterion):
+    def __init__(self, k: float = 3) -> None:
+        self.k = k
+
+    @property
+    def name(self) -> str:
+        return "Outlier Fraction Criterion"
+
+    def score(self, X: np.ndarray) -> float:
+        mu, sigma = np.mean(X), np.std(X)
+        return np.mean(np.abs(X - mu) > self.k * sigma)
+
+
+class CRange(ASampleClassifierCriterion):
+    @property
+    def name(self) -> str:
+        return "Range Criterion"
+
+    def score(self, X: np.ndarray) -> float:
+        return np.ptp(X)
+
+
+class CSkew(ASampleClassifierCriterion):
+    @property
+    def name(self) -> str:
+        return "Skewness Criterion"
+
+    def score(self, X: np.ndarray) -> float:
+        result = skew(X)
+        return result if not np.isnan(result) else 0
+
+
+class CSkewBowley(ASampleClassifierCriterion):
+    @property
+    def name(self) -> str:
+        return "Skewness Bowley Criterion"
+
+    def score(self, X: np.ndarray) -> float:
+        p25, p50, p75 = np.percentile(X, [25, 50, 75])
+        result = (p75 + p25 - 2 * p50) / (p75 - p25)
+        return result if not np.isnan(result) else 0
+
+
+class CSpacingGap(ASampleClassifierCriterion):
+    def __init__(self, rate: float = 5) -> None:
+        self.rate = rate
+
+    @property
+    def name(self) -> str:
+        return "Spacing Gap Criterion"
+
+    def score(self, X: np.ndarray) -> float:
+        diff = np.diff(np.sort(X))
+        dmedian = np.median(diff)
+        return np.mean(diff > self.rate * dmedian)
+
+
+class CSpacingGini(ASampleClassifierCriterion):
+    @property
+    def name(self) -> str:
+        return "Spacing Gini Criterion"
+
+    def score(self, X: np.ndarray) -> float:
+        diff = np.diff(np.sort(X))
+        diff = np.sort(diff)
+        n = len(diff)
+        index = np.arange(1, n + 1)
+        result = np.sum((2 * index - n - 1) * diff) / (np.sum(diff) * n)
+        return result if not np.isnan(result) else 0
+
+
+class CMaxZscore(ASampleClassifierCriterion):
+    @property
+    def name(self) -> str:
+        return "Z-Score Criterion"
+
+    def score(self, X: np.ndarray) -> float:
+        result = np.max(np.abs(zscore(X)))
+        return result if not np.isnan(result) else 0
+
+
+class CBootKurt(ASampleClassifierCriterion):
+    def __init__(self, n_boot: int = 200, state: int | None = None) -> None:
+        self.n_boot = n_boot
+        self.state = state
+
+    @property
+    def name(self) -> str:
+        return "Bootstrap Kurtosis Criterion"
+
+    def score(self, X: np.ndarray) -> float:
+        np.random.seed(self.state)
+
+        n = len(X)
+        means = [kurtosis(np.random.choice(X, size=n, replace=True)) for _ in range(self.n_boot)]
+        result = np.var(means)
+        return result if not np.isnan(result) else 0
```

Comment:
np.random.seed(self.state) sets the global random seed for NumPy. This means that if you run CBootKurt.score(), it will affect the outcome of every other random number generation in your entire program that uses np.random.
```suggestion
from numpy.random import Generator, PCG64

class CBootKurt(ASampleClassifierCriterion):
    def __init__(self, n_boot: int = 200, state: int | None = None) -> None:
        self.n_boot = n_boot
        self.rng = Generator(PCG64(state))

    @property
    def name(self) -> str:
        return "Bootstrap Kurtosis Criterion"

    def score(self, X: np.ndarray) -> float:
        n = len(X)
        
        means = [kurtosis(self.rng.choice(X, size=n, replace=True)) for _ in range(self.n_boot)]
        result = np.var(means)
        return result if not np.isnan(result) else 0
```